torchvision/transforms/functional_tensor.py

```diff
@@ -411,7 +413,22 @@ def pad(img: Tensor, padding: List[int], fill: int = 0, padding_mode: str = "con
         need_cast = True
         img = img.to(torch.float32)
 
-    img = torch_pad(img, p, mode=padding_mode, value=float(fill))
+    img = torch_pad(img, p, mode=padding_mode)
```

Comment:
if `fill` is a scalar now we still transform it to a tensor and apply to the image at most 4 times below (`img[..., :, :pad_left] = fill_img`). Maybe, for performance reasons we could do if/else here and keep previous behaviour with a single torch_pad call for scalars and for list/tuple do what you coded ?

Reply:
I had that, see earlier versions of the commit. Unfortunately I couldn't find a way to write it in a JIT-friendly way. See [here](https://github.com/pytorch/vision/pull/5568/commits/ccef4e4c58d4a344f4fa86178472806dccf72d9c#diff-497b983ae7c82237c4a0722bc6f6637525e0cdf8369dbd78ee19cf986bffe258R414) for more details. If you have ideas on how to have the optimization and be JIT-scriptable I'm happy to use them :)